src/util.rs

```diff
@@ -86,3 +86,18 @@ pub const fn weeks_in_year(year: i32) -> u8 {
         52
     }
 }
+
+pub(crate) trait Clamp {
+    /// Constrain `self` between `min` and `max` (inclusive).
+    ///
+    /// If `self` is less than `min`, returns `min`.
+    /// If `self` is greater than `max`, returns `max`.
+    /// Otherwise, returns `self`.
+    fn clamp(self, min: Self, max: Self) -> Self;
+}
+
+impl<T: Ord> Clamp for T {
+    fn clamp(self, min: Self, max: Self) -> Self {
+        std::cmp::max(min, std::cmp::min(self, max))
```

Comment:
This should use `core` rather than `std`, and is why CI is failing.

Reply:
Done!